reana_server/scheduler.py

```diff
@@ -64,22 +65,28 @@ def check_predefined_conditions():
 
 def doesnt_exceed_max_reana_workflow_count():
     """Check upper limit on running REANA batch workflows."""
+    doesnt_exceed = True
     try:
-        running_workflows = Workflow.query.filter(
-            or_(
-                Workflow.status == RunStatus.pending,
-                Workflow.status == RunStatus.running,
+        running_workflows = (
+            Session.query(func.count())
+            .filter(
+                or_(
+                    Workflow.status == RunStatus.pending,
+                    Workflow.status == RunStatus.running,
+                )
             )
-        ).count()
+            .scalar()
+        )
         if running_workflows >= REANA_MAX_CONCURRENT_BATCH_WORKFLOWS:
-            return False
+            doesnt_exceed = False
     except SQLAlchemyError as e:
         logging.error(
             "Something went wrong while querying for number of running workflows."
         )
         logging.error(e)
-        return False
-    return True
+        doesnt_exceed = False
+    Session.commit()
```

Comment:
Why `Session.commit()` is needed here? Since we are not mutating the query, isn't Session.close() enough in this case?

Reply:
In normal conditions, nothing should be needed... I guess in this context `commit` has the same behavior as `closes`. It'd be nice to know why it's needed..

Reply:
I thought about commit mostly because of what might happen outside of this function by the callers. If this function gets used during long chain of calls/contexts, the session may perhaps be used later, so a commit rather than a flat-out close seemed "safer". (That said, if we revive `pool_pre_ping=True` option to the DB engine, for other reasons, then we could do either...)